### PR TITLE
feat: store distance scores in scores dict

### PR DESCRIPTION
**Pull request by anna-charlotte**
Signed-off-by: anna-charlotte <charlotte.gerhaher@jina.ai>

Goals:
Milvus: Store distance scores in a `Document`'s `.scores` dictionary, under the key `'score'` to enable accessing those scores.

- [x] store distance values in `.scores` dict with `key = 'score'`
- [x] update milvus documentation
- [x] update 'Add New Document Store' section in documentation

### CODE REVIEW

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

File: docarray/array/storage/milvus/backend.py [hunk 1]
```diff
@@ -271,11 +272,12 @@ def _docs_from_search_response(
...
         for r in responses:
-            das.append(
-                DocumentArray(
-                    [Document.from_base64(hit.entity.get('serialized')) for hit in r]
-                )
-            )
+            da = []
+            for hit in r:
+                doc = Document.from_base64(hit.entity.get('serialized'))
+                doc.scores[f'score'] = NamedScore(value=hit.score)
```
Comment (JohannesMessner) [inline docarray/array/storage/milvus/backend.py:278]: Is there a way we can know the score that was used? I think it is in the config, can we use that as the key here? how do other backends handle that?

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

File: docs/advanced/document-store/milvus.md [hunk 2]
```diff
@@ -253,6 +253,25 @@ Embeddings Nearest Neighbours with "price" at most 7:
...
+````{admonition} Note
+:class: note
+For Milvus, the distance scores can be accessed in the Document's `.scores` dictionary under the key `'score'`.
+````
+You can access the scores as follows:
+
+````python
+for doc in results:
+    print(f"score = {doc.scores['score'].value}")
+````
```
Comment (JohannesMessner) [inline docs/advanced/document-store/milvus.md:266]: this should be adapted to the new key in the scores dict

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

**Review by JohannesMessner — APPROVED** [review]
LGTM! There is one test failing unexpectedly, should not be related, but let's restart the CI before we merge

**Pull request merged**